department-of-veterans-affairs/caseflow
# CASEFLOW-486 Add judges and attorneys to those that can edit DOD date

Description
* The ability to edit NOD dates was previously restricted to BvaIntake.singleton.users and ClerkOfTheBoard.singletone.users. 
* Permissions have been expanded to include attorneys and judges.
* Displaying the "Edit NOD Date" link in the Queue Timeline section needs to be restricted to attorneys in addition to BvaIntake.singleton.users and ClerkOfTheBoard.singletone.users.

Acceptance Criteria
 * Code compiles correctly
 * NOD exists and the "Edit NOD Date" link displays for Attorneys, Judges, ClerkOfTheBoard, and BvaIntake users.
  * NOD exists and the "Edit NOD Date" link does not display for other user types.
 
Testing Plan
* Login to Queue as each of the following:  COB_User (Clerk of the Board user), BVAOSHOWAL (judge), BVACASPER1 (attorney), BVASHAW (Intake)
* Make sure "edit_nod_date" featuretoggle is enabled.
* Choose an appeal that has an existing NOD (123456789) and confirm that the link displays.
* Switch user to someone without permission (BVAGBLACK)
Choose an appeal that has an existing NOD and confirm that the link does not display (123456789)

User Facing Changes
* See screenshots from [#15672 ](https://github.com/department-of-veterans-affairs/caseflow/pull/15672)
* No additional changes from this PR
